File: back/src/main/java/com/tdevilleduc/urthehero/back/config/SwaggerConfig.java
package com.tdevilleduc.urthehero.back.config;

import com.google.common.base.Predicate;
import com.google.common.base.Predicates;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.PropertySource;
import springfox.documentation.RequestHandler;
import springfox.documentation.builders.PathSelectors;
import springfox.documentation.builders.RequestHandlerSelectors;
import springfox.documentation.service.ApiInfo;
import springfox.documentation.service.Contact;
import springfox.documentation.spi.DocumentationType;
import springfox.documentation.spring.web.plugins.Docket;
import springfox.documentation.swagger2.annotations.EnableSwagger2;

import java.util.Collections;


@Configuration
@EnableSwagger2
@PropertySource(value = {
        "classpath:/swagger/swagger-messages_fr.properties"
})
class SwaggerConfig {
    @Bean
    public Docket api() {
        return new Docket(DocumentationType.SWAGGER_2)
                .select()
                .apis(apis())
                .paths(paths())
                .build()
                .apiInfo(apiInfo());
    }

    private Predicate<String> paths() {
        return Predicates.not(PathSelectors.regex("/basic-error-controller.*"));
    }

    private Predicate<RequestHandler> apis() {
        return Predicates.not(RequestHandlerSelectors.basePackage("org.springframework"));
    }

    private ApiInfo apiInfo() {
        return new ApiInfo(
                "URtheHero API",
                "BackEnd API for URtheHero application.",
                "API 0.0.1-SNAPSHOT",
                "Terms of service",
                new Contact("Thomas Devile-Duc", "urthehero.tdevilleduc.com", "urthehero.app@gmail.com"),
                "License of API", "API license URL", Collections.emptyList());
    }
}

File: back/src/main/java/com/tdevilleduc/urthehero/back/config/SwaggerConfig.kt
package com.tdevilleduc.urthehero.back.config

import io.swagger.v3.oas.annotations.enums.SecuritySchemeType
import io.swagger.v3.oas.annotations.security.SecurityScheme
import io.swagger.v3.oas.models.ExternalDocumentation
import io.swagger.v3.oas.models.OpenAPI
import io.swagger.v3.oas.models.info.Info
import io.swagger.v3.oas.models.info.License
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.PropertySource

@Configuration
@PropertySource(value = [
    "classpath:/swagger/swagger-messages_fr.properties",
    "classpath:/swagger/swagger-model-messages_fr.properties"
])
@SecurityScheme(
        name = "bearerAuth",
        type = SecuritySchemeType.HTTP,
        bearerFormat = "JWT",
        scheme = "bearer"
)
internal class SwaggerConfig {

    /**
     * Create default ApiInfo
     * @param properties
     * @return
     */
    @Bean
    fun springShopOpenAPI(): OpenAPI? {
        return OpenAPI()
                .info(Info().title("SpringShop API")
                        .description("Spring shop sample application")
                        .version("v0.0.1")
                        .license(License().name("Apache 2.0").url("http://springdoc.org")))
                .externalDocs(ExternalDocumentation()
                        .description("SpringShop Wiki Documentation")
                        .url("https://springshop.wiki.github.org/docs"))
    }

}